spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/RddSelectPdTrf.scala

```diff
@@ -0,0 +1,153 @@
+      case _: In | _: InSet =>      "IN"
+      case _ => throw new UnsupportedOperationException(
+        "It's not a valid predicate to be pushed down, only >, <, >=, <= and In are allowed: " + predicate)
+    }
+  }
+
+  override protected def predicateToCql(predicate: Any): String = {
+    predicate match {
+      case cmp: BinaryComparison =>
+        cmp.references.head.name + " " + predicateOperator(cmp) + " ?"
+      case in: In =>
+        in.value.references.head.name + " IN " + in.list.map(_ => "?").mkString("(", ", ", ")")
+      case inset: InSet =>
+        inset.value.references.head.name + " IN " + inset.hset.toSeq.map(_ => "?").mkString("(", ", ", ")")
+      case _ =>
+        throw new UnsupportedOperationException(
+          "It's not a valid predicate to be pushed down, only >, <, >=, <= and In are allowed: " + predicate)
+    }
+  }
+
+  private[this] def castFromString(value: String, dataType: DataType) = Cast(Literal(value), dataType).eval(null)
+
+}
+
+/**
+ * Transform a RDD by filtering columns and pushdown source [[Filter]]s
+ */
+class RddFilterSelectPdTrf(requiredColumns: Array[String], columnNameByLowercase: Map[String, String],
+                           filters: Seq[Filter]) extends RddSelectPdTrf {
+
+  override var maybeSelect = if (requiredColumns.nonEmpty) { rdd: RDDType =>
+    rdd.select(requiredColumns.map(a => columnNameByLowercase(a): NamedColumnRef): _*)
+  } else { rdd: RDDType =>
+    rdd
+  }
+
+  override protected var maybePushdownPredicates = whereClause(filters) match {
+    case (cql, values) if values.nonEmpty => rdd: RDDType => rdd.where(cql, values: _*)
+    case _ => rdd: RDDType => rdd
+  }
+
+
+  override protected def predicateRhsValue(filter: Any): Seq[Any] = {
+    filter match {
+      case eq: sources.EqualTo            => Seq(eq.value)
+      case lt: sources.LessThan           => Seq(lt.value)
+      case le: sources.LessThanOrEqual    => Seq(le.value)
+      case gt: sources.GreaterThan        => Seq(gt.value)
+      case ge: sources.GreaterThanOrEqual => Seq(ge.value)
+      case in: sources.In                 => in.values.toSeq
+      case _ => throw new UnsupportedOperationException("Unsupported filter: " + filter)
+    }
+  }
+
+  private[this] def filterOperator(filter: Filter): String = {
+    filter match {
+      case _: sources.EqualTo            => "="
+      case _: sources.LessThan           => "<"
+      case _: sources.LessThanOrEqual    => "<="
+      case _: sources.GreaterThan        => ">"
+      case _: sources.GreaterThanOrEqual => ">="
+      case _: sources.In                 => "IN"
+      case _ => throw new UnsupportedOperationException(
+        "It's not a valid filter to be pushed down, only >, <, >=, <= and In are allowed: " + filter)
+    }
+  }
+
+  override protected def predicateToCql(filter: Any): String = {
+    filter match {
+      case eq: sources.EqualTo            => eq.attribute + " " + filterOperator(eq) + " ?"
```

Comment:
This could be written in more Scala'ish style: `case EqualTo(attribute, _) => s" $attribute = ? "`.